fabric-combat-api-v1/src/main/java/net/fabricmc/fabric/api/combat/v1/ShotProjectileEvents.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.combat.v1;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public interface ShotProjectileEvents {
+	Event<ProjectileFromBow> BOW_SHOT_PROJECTILE = EventFactory.createArrayBacked(ProjectileFromBow.class, callbacks -> (ItemStack bowStack, ItemStack arrowStack, LivingEntity user, float pullProgress, PersistentProjectileEntity persistentProjectileEntity) -> {
```

Comment:
There may be a usage issue here, so the way it looks in rest of codebase, the custom bow logic is implemented via this event directly on the bow item. However this can cause an issue where a custom bow launches a rock and then a listener replaces said rock with a poisoned arrow.

Essentially this has an issue where the last mod to be registered wins.

Reply:
Yeah, im not sure how to get around this issue